Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -243,11 +247,37 @@ def run_checks(
           )
         end
 
+        # Check current and latest resources (if "--resources" flag is given)
+        # Only check current and latest versions if we have resources to check against
+        check_for_resources = check_resources && formula_or_cask.is_a?(Formula) && formula_or_cask.resources.present?
+        if check_for_resources
+          resource_version_info = formula_or_cask.resources.map do |resource|
+            res_skip_info ||= SkipConditions.skip_information(resource, verbose: verbose)
+            if res_skip_info.present?
+              res_skip_info
+            else
+              res_version_info = resource_version(
+                resource,
+                json:    json,
+                debug:   debug,
+                quiet:   quiet,
+                verbose: verbose,
+              )
+              if res_version_info.empty?
+                status_hash(resource, "error", ["Unable to get versions"], verbose: verbose)
+              else
+                res_version_info
+              end
+            end
+          end.compact_blank
+        end
+
```

Comment:
This code should come after the main formula checks, as resource checks are secondary to the main formula check. Currently, because this code comes before the main check, the debug output shows the resource information before the formula information, which isn't correct. For example, with my modified version of `influxdb` I'm seeing:

```
Resource:         pkg-config-wrapper
Livecheckable?:   Yes

URL (url):        https://github.com/influxdata/pkg-config/archive/refs/tags/v0.2.11.tar.gz
URL (processed):  https://github.com/influxdata/pkg-config.git
Strategy:         Git
Regex:            /^v?(\d+(?:\.\d+)+)$/i

Matched Versions:
0.2.1, 0.1.0, 0.1.1, 0.2.0, 0.2.10, 0.2.11, 0.2.12, 0.2.2, 0.2.3, 0.2.4, 0.2.5, 0.2.6, 0.2.7, 0.2.8, 0.2.9
Formula:          influxdb
Livecheckable?:   Yes

URL (stable):     https://github.com/influxdata/influxdb.git
Strategy:         Git
Regex:            /^v?((?!9\.9\.9)\d+(?:\.\d+)+)$/i

Matched Versions:
1.8.0, 0.0.1, 0.0.2, 0.0.3, 0.0.4, 0.0.5, 0.0.6, 0.0.7, 0.0.8, 0.0.9, 0.1.0, 0.10.0, 0.10.1, 0.10.2, 0.10.3, 0.11.0, 0.11.1, 0.12.0, 0.12.1, 0.12.2, 0.13.0, 0.2.0, 0.3.0, 0.3.1, 0.3.2, 0.4.0, 0.4.1, 0.4.2, 0.4.3, 0.4.4, 0.5.0, 0.5.1, 0.5.10, 0.5.11, 0.5.12, 0.5.2, 0.5.3, 0.5.4, 0.5.5, 0.5.6, 0.5.7, 0.5.8, 0.5.9, 0.6.0, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.6.5, 0.7.0, 0.7.1, 0.7.2, 0.7.3, 0.8.0, 0.8.1, 0.8.2, 0.8.3, 0.8.4, 0.8.5, 0.8.6, 0.8.7, 0.8.8, 0.9.0, 0.9.1, 0.9.2, 0.9.2.1, 0.9.3, 0.9.4, 0.9.4.1, 0.9.4.2, 0.9.5, 0.9.5.1, 0.9.6, 0.9.6.1, 1.0.0, 1.0.1, 1.0.2, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.1.4, 1.1.5, 1.10.0, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.8, 1.3.9, 1.4.0, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.5.5, 1.6.0, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.7.0, 1.7.1, 1.7.10, 1.7.11, 1.7.2, 1.7.3, 1.7.4, 1.7.5, 1.7.6, 1.7.7, 1.7.8, 1.7.9, 1.8.1, 1.8.10, 1.8.2, 1.8.3, 1.8.4, 1.8.5, 1.8.6, 1.8.7, 1.8.8, 1.8.9, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.9.4, 1.9.5, 1.9.6, 1.9.7, 1.9.8, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.0.8, 2.0.9, 2.1.0, 2.1.1, 2.2.0, 2.3.0, 2.4.0

influxdb: 2.4.0 ==> 2.4.0
  pkg-config-wrapper: 0.2.11 ==> 0.2.12
  ui-assets: skipped - Update manually
```

Reply:
I moved this before the `SkipConditions` logic for `Formula` as per @nandahkrishna's suggestion. i.e: we should still get the resources' info in case `Formula` was skipped.

Reply:
@mohammadzainabbas I believe there's a misunderstanding here. I think:
* If a Formula is skipped, its resources don't matter (we can ignore checking them).
* If a Formula's `livecheck` errors out, we should still attempt to show resource versions.

The resource outputs should always be printed after the formula output.

Reply:
Oh okay, I have updated the code accordingly.